### DISCUSSION TITLE
Installing without running post-install scripts?

### DISCUSSION THREAD

**Discussion by benrhere** [plouc/nivo#1887] [Q&A] [answered]
Hi,

I've been trying to run `npm install` with `--ignore-scripts` after reading about some high profile vendor attacks where an NPM package was compromised and the post-install script did some malicious things. But whenever I'm upgrading nivo packages, after updating dependences when I run
`npm install --ignore-scripts`
(bumping bar and pie from 0.78.0 to 0.79.1 in this example, and core to 0.79.0)
I get:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: [package]@1.0.0
npm ERR! Found: @nivo/bar@0.78.0
npm ERR! node_modules/@nivo/bar
npm ERR!   @nivo/bar@"0.79.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! @nivo/bar@"0.79.1" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: @nivo/core@0.79.0
npm ERR! node_modules/@nivo/core
npm ERR!   peer @nivo/core@"0.79.0" from @nivo/bar@0.79.1
npm ERR!   node_modules/@nivo/bar
npm ERR!     @nivo/bar@"0.79.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See ...eresolve-report.txt for a full report.

I've tried wiping the package-lock.json, and wiping the node_modules folder. Neither action seems to help. But running with post-install scripts works without error.

Does nivo currently rely on a post-install script? If so, could you point me toward where it lives in the repo so I could run those steps manually when updating?

Thanks! As well, thanks for all the work you do on this package - it's terrific.

**Comment by plouc** [comment, marked as answer]
Hi @benrhere,

No, nivo doesn't rely on any post-install script, so I guess the issue comes from other packages.

**Reply by benrhere** [reply]
Thanks @plouc , I suppose I'll have to dig deeper.

**Reply by benrhere** [reply]
I dug a bit deeper and found the problem on my side. In addition to a ".package-lock.json" located at the top level, there happened to be another similar file within the node_modules directory. I wiped this out, but still saw the error. I then went into the @nivo directory and found that it contained 0.78 versions of some assets. I removed the directory and reran with --ignore-scripts, and it worked. Thanks!

**Reply by plouc** [reply]
I'm glad you solved your issue :)
